src/UKHO.S100PermitService.Common/Encryption/S100Crypt.cs

```diff
@@ -35,5 +43,33 @@ public IEnumerable<ProductKey> GetDecryptedKeysFromProductKeys(IEnumerable<Produ
 
             return productKeys;
         }
+
+        public List<UpnInfo> GetDecryptedHardwareIdFromUserPermit(UserPermitServiceResponse userPermitServiceResponse)
+        {
+            _logger.LogInformation(EventIds.GetHwIdFromUserPermitStarted.ToEventId(), "Get decrypted hardware id from user permits started");
+
+            List<UpnInfo> listOfUpnInfo = [];
```

Comment:
Used new List<UpnInfo>() instead of [] for list initialization.

```suggestion
            var listOfUpnInfo = new List<UpnInfo>();
```

Reply:
Done